sdk/confidentialledger/azure-resourcemanager-confidentialledger/src/test/java/com/azure/resourcemanager/confidentialledger/ConfidentialLedgerManagementTestBase.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.confidentialledger;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.core.test.TestBase;
+import com.azure.core.test.TestMode;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+
+import com.azure.resourcemanager.resources.ResourceManager;
+import com.azure.resourcemanager.resources.models.ResourceGroup;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConfidentialLedgerManagementTestBase extends TestBase {
+    private static AzureProfile azureProfile;
+    private static TokenCredential credential;
+    private static ResourceGroup testResourceGroup;
+    private ConfidentialLedgerManagementOperations ledgerOperationsInstance;
+    @BeforeAll
+    public static void setup() {
+        // Authenticate
+        setAzureProfile();
+        setCredential();
+
+        // Create a resource group for testing in LIVE and RECORD modes only
+        String testResourceGroupName = "acl-sdk-test-rg";
+        setTestResourceGroup(testResourceGroupName);
+    }
+    @AfterAll
+    public static void cleanUp() {
+        // If AZURE_TEST_MODE isn't set to a value, use PLAYBACK mode as a default
+        String testMode = System.getenv("AZURE_TEST_MODE");
+        if (testMode == null) {
+            testMode = "PLAYBACK";
+        }
+
+        // Delete the created resource group in LIVE and RECORD modes only
+        if (!("PLAYBACK".equals(testMode))) {
+            ResourceManager
+                .authenticate(getCredential(), getAzureProfile())
+                .withDefaultSubscription()
+                .resourceGroups()
+                .deleteByName(testResourceGroup.name());
+        }
+    }
+    @BeforeEach
+    public void setupManager() {
+        ConfidentialLedgerManager ledgerManager = null;
+        if (getTestMode() == TestMode.LIVE) {
+            ledgerManager = ConfidentialLedgerManager
+                .authenticate(getCredential(), getAzureProfile());
+        } else if (getTestMode() == TestMode.RECORD) {
+            ledgerManager = ConfidentialLedgerManager
+                .configure()
+                .withPolicy(interceptorManager.getRecordPolicy())
+                .authenticate(getCredential(), getAzureProfile());
+        } else if (getTestMode() == TestMode.PLAYBACK) {
+            ledgerManager = ConfidentialLedgerManager
+                .configure()
+                .withHttpClient(interceptorManager.getPlaybackClient())
+                .authenticate(getCredential(), getAzureProfile());
```

Comment:
you need to put the subscriptionId to AzureProfile in PLAYBACK, as there is no env variable for it.

You may use `new AzureProfile(null, "027da7f8-2fc6-46d4-9be9-560706b60fec", AzureEnvironment.AZURE)` (that is your recorded subscriptionId).

PS: That is the reason I don't like RECORD/PLAYBACK tests.

Reply:
Is there a way to mock the authentication part in the PLAYBACK mode?

Reply:
Yes, I understand this part about the subscription ID needed and I added it in the code. What I mean is that the CI check is still failing, and I think it's due to not being able to authenticate because of as you said there are no environment variables in CI. So, I'm asking how to solve this issue?

Reply:
I see. Please do not use the `DefaultAzureCredential`. Use a mocked credential like this https://github.com/Azure/azure-sdk-for-java/blob/b9a185753539828b13fdce1aa91427b0aa997898/sdk/devcenter/azure-developer-devcenter/src/test/java/com/azure/developer/devcenter/DevCenterClientTestBase.java#L38